src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs

```diff
@@ -103,7 +103,7 @@ private async Task ApplyImplementationAsync(BuildShapeContext context, string di
                 return;
             }
 
-            var newShape = await _shapeBuilder(context);
+            var newShape = Shape = await _shapeBuilder(context);
```

Comment:
Is it a typo @sebastienros ?
`var newShape = Shape = await _shapeBuilder(context);`